yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -988,41 +943,49 @@ func (p *ParserIssueDetector) SetColumnStatistics(columnStats []utils.ColumnStat
 	}
 }
 
-// addIndexToCoverage adds an index and its prefixes to the coverage map for checking missing foreign key index issue
+// addIndexToCoverage adds an index to the table indexes map for checking missing foreign key index issue
 func (p *ParserIssueDetector) addIndexToCoverage(index *queryparser.Index) {
 	tableName := index.GetTableName()
 
-	// Extract columns in order (only non-expression columns)
-	columns := make([]string, 0)
-	for _, param := range index.Params {
-		if !param.IsExpression {
-			columns = append(columns, param.ColName)
+	// Add the index to the table's index list
+	if _, exists := p.tableIndexes[tableName]; !exists {
+		p.tableIndexes[tableName] = make([]*queryparser.Index, 0)
+	}
+	p.tableIndexes[tableName] = append(p.tableIndexes[tableName], index)
+}
+
+// addConstraintAsIndex creates a mock index object from a constraint and adds it to the table indexes map.
+// This is used for primary keys and unique constraints which are also indexes for missing foreign key index detection.
+func (p *ParserIssueDetector) addConstraintAsIndex(tableName string, columns []string, constraintType string) {
```

Comment:
Nice. I think this is a good idea 👍  

@priyanshi-yb can you pls take a look at this function alone to review that we are populating the struct properly. ?